src/erlcloud_firehose_impl.erl

```diff
@@ -0,0 +1,177 @@
+%% -*- mode: erlang;erlang-indent-level: 4;indent-tabs-mode: nil -*-
```

Comment:
Sorry for the delay, I added the erlcloud_firehose module. I'm going to go back and refactor the _impl into the module so we don't have the duplicate impl for kinesis and firehose.